lib/Google/RestApi/Auth/ServiceAccount.pm

```diff
@@ -0,0 +1,116 @@
+package Google::RestApi::Auth::ServiceAccount;
+
+use strict;
+use warnings;
+
+our $VERSION = '0.3';
+
+use 5.010_000;
+
+use autodie;
+use File::Basename;
+use Type::Params qw(compile_named);
+use Types::Standard qw(Str ArrayRef);
+use WWW::Google::Cloud::Auth::ServiceAccount;
+use YAML::Any qw(Dump);
+
+no autovivification;
+
+use Google::RestApi::Utils qw(config_file);
+
+use parent 'Google::RestApi::Auth';
+
+do 'Google/RestApi/logger_init.pl';
+
+sub new {
+  my $class = shift;
+
+  my $self = config_file(@_);
+  state $check = compile_named(
+    config_file        => Str, { optional => 1 },
+    parent_config_file => Str, { optional => 1 },  # only used internally
+    account_file       => Str,
+    scope              => ArrayRef[Str],
+  );
+  $self = $check->(%$self);
+
+  return bless $self, $class;
+}
+
+sub headers {
+  my $self = shift;
+  return $self->{headers} if $self->{headers};
```

Comment:
this may cause issues with long running scripts since the token may expire.
WWW::Google::Cloud::Auth::ServiceAccount already takes care of caching and requesting a new token if the present one has expired.

Reply:
OK, this was just a roughed-out suggestion. I've only had time to look into service accounts very superficially.